### PR TITLE
Fix IsNonBundledAssemblyLoadingSupported condition in PlatformDetection

### PR DESCRIPTION
**Pull request by steveisok**
Inadvertently put in a condition that only made tests with this check run in NativeAOT. The intent was to skip on both NativeAOT and Mono AOT.

### CODE REVIEW

File: src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs
Comment (stephentoub) [inline src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs:156]: I'm curious what's different between Native AOT and Mono's AOT that the former is able to support loading assemblies from files and the latter isn't?

**Review by stephentoub — APPROVED** [review]

**Review by mdh1418 — COMMENTED** [review]
The thing that isn't supported, is it specifically loading an assembly from a file on mono, or is it the assembly execution from a file on mono that is not supported? Isn't this AssemblyLoading from a file as well https://github.com/dotnet/runtime/blob/1e7299db2b6f25dc033d69c9f9e8d2ddc5126db8/src/libraries/System.Reflection/tests/AssemblyTests.cs#L348?

File: src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs
Comment (steveisok) [reply src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs]: In full AOT for mono, assembly loading is allowed, but in the case I'm trying to skip, the exe assembly wasn't AOT'd and was not registered as a module on startup.  For NativeAOT, my understanding is everything compiles into a single binary. That means you can't load anything external.

I kept going back and forth on a proper name for the property.  I reverted back to what I originally had as I now think it's a little more accurate.

**Comment by steveisok** [comment]
@vargaz These are still failing because it's trying to JIT:

https://github.com/dotnet/runtime/blob/aac729ff906a31f327823587748687c0308a4043/src/libraries/System.Runtime.InteropServices/tests/System.Runtime.InteropServices.UnitTests/System/Runtime/InteropServices/Marshal/PtrToStructureTests.cs#L153-L160

https://github.com/dotnet/runtime/blob/aac729ff906a31f327823587748687c0308a4043/src/libraries/System.Runtime.InteropServices/tests/System.Runtime.InteropServices.UnitTests/System/Runtime/InteropServices/Marshal/PtrToStructureTests.cs#L164-L171

https://github.com/dotnet/runtime/blob/aac729ff906a31f327823587748687c0308a4043/src/libraries/System.Runtime.InteropServices/tests/System.Runtime.InteropServices.UnitTests/System/Runtime/InteropServices/Marshal/StructureToPtrTests.cs#L261-L282

File: src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs
Comment (MichalStrehovsky) [reply src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs]: In NativeAOT, Assembly.Load works too because that one is not actually loading anything new. The assembly was part of the app.

Assembly.LoadFrom or loading from byte array don't work and are blocked on the existing condition. Does it match your semantic?

Comment (steveisok) [reply src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs]: Close. I say that because tests like this pass:

https://github.com/dotnet/runtime/blob/46db4aa3c505ac36cf8d0d612d4161bb7239ed87/src/libraries/System.Runtime/tests/System/ActivatorTests.cs#L626-L648

Comment (MichalStrehovsky) [reply src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs]: I'll leave it up to you, but I would expect that API to be unsupported in general. It might work in corner cases like in the test but it's a much easier story to say to customers that apis that load assemblies from files are not supported than to have an asterisk with when it is supported. Especially if there's perfectly good replacement apis like CreateInstance(string,string).

Apis that are unsupported don't need test coverage.

Comment (steveisok) [reply src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs]: Tagging @lambdageek in the event I didn't explain it quite right.

**Comment by MichalStrehovsky** [comment]
> The thing that isn't supported, is it specifically loading an assembly from a file on mono, or is it the assembly execution from a file on mono that is not supported? Isn't this AssemblyLoading from a file as well

This one is in the same category as the `Activator.CreateInstanceFrom` case above - it's loading a file, but the file happens to be part of the app, so it probably works.

The problem (and test hole) is that the tests are not testing the primary customer scenario for various `LoadFrom` APIs - the APIs are meant for e.g. a plugin that wasn't part of the app.

If we were to adapt the tests to test the primary customer scenarios and not the corner case that happens to work (the loaded assembly was part of the app already), annotating these tests as `PlatformDetection.IsAssemblyLoadingSupported` and saying that loading assemblies from files is not supported on Mono AOT (even though it might work, but it's not tested) would be a good fix. I'm not sure introducing new `PlatformDetection` just to keep test coverage for what is actually a non-mainstream scenario has a lot of value.

But don't consider this feedback blocking. What is here works too.

**Pull request merged**
